packages/language-server/src/plugins/typescript/features/FindFileReferencesProvider.ts

```diff
@@ -0,0 +1,12 @@
+import { Location } from 'vscode-languageserver';
+import { Document } from '../../../lib/documents';
+import { FileReferencesProvider } from '../../interfaces';
+import { LSAndTSDocResolver } from '../LSAndTSDocResolver';
+
+export class FindFileReferencesProviderImpl implements FileReferencesProvider {
+    constructor(private readonly lsAndTsDocResolver: LSAndTSDocResolver) {}
+
+    async fileReferences(fileName: string): Promise<Location[] | null> {
```

Comment:
Is this where we talk to tsserver?

Reply:
ts language service. It would be mostly the same as `FindReferenceProvider`. Just change the `getReferencesAtPosition` to `getFileReferences` and remove this line

https://github.com/sveltejs/language-tools/blob/44d2d65132616143cff5124cd0914832494dd513/packages/language-server/src/plugins/typescript/features/FindReferencesProvider.ts#L34

Reply:
I added what you mentioned but a bit unsure of         const { lang, tsDoc } = await this.getLSAndTSDoc(document);
which requires a "document". What is document? Is it necessary to be passing through all the calls in order to get the tsDoc as FindReferencesProviderImpl does?

Still unable to debug this so it's pretty black box atm

Reply:
Yes, it should be a Document. The argument from the client should be a URI. And we'll use the URI to get the document in the `PluginHost`. 

I looked into it and think the context menu part is a little bit hard to explain. So I added some code for the extension side. You can now see the command in the context menu of the file tab.